### PR TITLE
92 cluster make sure cluster stays online during all tests

### LINKED ISSUE #92: Cluster: make sure cluster stays online during all tests
There should be zero downtime.

### PR DESCRIPTION
**Pull request by MuhammadQadora**
Fix #92

Tested Reads and writes at failover time to make sure we have zero downtime.

## Summary by CodeRabbit

- **New Features**
	- Introduced additional job configurations for zero-downtime testing of clusters on Google Cloud Platform.
	- Added a comprehensive testing framework to validate zero-downtime capabilities and failover scenarios for Omnistrate fleet instances.

- **Bug Fixes**
	- Enhanced testing capabilities to ensure data integrity during failover events.
	- Increased deployment creation timeout to accommodate longer processes during testing.
	- Implemented a retry mechanism for establishing connections to improve resilience.

- **Chores**
	- Updated command-line argument parsing for improved readability in testing scripts.
	- Improved formatting in scripts for better readability.
	- Updated dependency specifications for enhanced stability and maintainability.

### CODE REVIEW

**Comment by MuhammadQadora** [comment]
Hi,
for the most significant part (finding) of this test, is that the failover takes place almost instantaneously. My expected behaviour was that the "DEPLOYING" state would last about a minute. I tested using the while loop to see if there is a tiny state error where the state goes from "DEPLOYING" to "RUNNING" for a few seconds and then "DEPLOYING" again, but it stays in the RUNNING state.

File: .github/workflows/build-test-image.yaml [hunk 1]
```diff
@@ -309,6 +309,26 @@ jobs:
...
+          - name: PRO/ClusterMultiZone - GCP/us-central1 - Test cluster zero downtime
+            if: "true"
+            testFile: test_cluster_zerodowntime.py
+            tierName: pro-${{ contains(github.ref, 'refs/tags/v') && 'main' || github.ref_name }}
+            cloudProvider: gcp
+            cloudRegion: us-central1
+            subscriptionId: sub-GJPV3NoNC0
+            serviceId: ${{ vars.OMNISTRATE_INTERNAL_SERVICE_ID }}
+            environmentId: ${{ vars.OMNISTRATE_INTERNAL_DEV_ENVIRONMENT}}
+            extraParams: "--resource-key 'cluster-Multi-Zone' --replica-id 'cluster-mz-0' --instance-name 'test-cluster-mz-zero-downtime' --instance-description 'test-cluster-mz-zero-downtime' --instance-type 'e2-custom-4-8192' --storage-size '30' --rdb-config 'medium' --aof-config 'always' --host-count '6' --cluster-replicas '1'"
```
Comment (dudizimber) [inline .github/workflows/build-test-image.yaml:331]: Why do we need this extra test? I think we discussed implementing the zero downtime during the tests already written, no?

File: omnistrate_tests/test_cluster_zerodowntime.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        # Test failover and data loss
+        test_failover(instance)
+    except Exception as e:
+        instance.delete(True)
+        raise e
+
+    # Delete instance
+    instance.delete(True)
+
+    print("Test passed")
```
[the remark this answers is not in the document]
Comment (MuhammadQadora) [reply omnistrate_tests/test_cluster_zerodowntime.py]: not sure if we should add a logger. till now @dudizimber did not use one.

**Review by dudizimber — APPROVED** [review]
